docs/module.rst

```diff
@@ -0,0 +1,267 @@
+
+Nextflow implicitly looks for the script file ``./some/module.nf``, resolving the path
+against the *including* script location.
+
+Module includes are subject to the following rules:
+
+- Relative paths must begin with the ``./`` prefix.
+- Include statements are not allowed from within a worklfow. They must occur at the script level.
+
+.. _module-directory:
+
+Module directory
+----------------
+
+.. note::
+    This feature requires Nextflow version ``22.10.0`` or later.
+
+A module can be defined as a directory with the same name as the module and with a script
+named ``main.nf``. For example::
+
+    some
+        └-module
+            └-main.nf
+
+When defined as a directory, the module must be included by specifying the module directory path::
+
+    include { foo } from './some/module'
+
+Module directories allow the use of module scoped binaries scripts. See `Module binaries`_ for details.
+
+Multiple inclusions
+-------------------
+
+A Nextflow script can include any number of modules, and an ``include`` statement can import any number of
+components from a module. Multiple components can be included from the same module by using the syntax
+shown below::
+
+    include { foo; bar } from './some/module'
+
+    workflow {
+        data = channel.fromPath('/some/data/*.txt')
+        foo(data)
+        bar(data)
+    }
+
+
+.. _module-aliases:
+
+Module aliases
+--------------
+
+When including a module component, it's possible to specify an *alias* with the ``as`` keyword.
+Aliasing allows you to include and invoke components with the same name, by assigning them different
+names in the including context. For example::
+
+    include { foo } from './some/module'
+    include { foo as bar } from './other/module'
+
+    workflow {
+        foo(some_data)
+        bar(other_data)
+    }
+
+You can even include the same component multiple times under different names::
+
+    include { foo; foo as bar } from './some/module'
+
+    workflow {
+        foo(some_data)
+        bar(other_data)
+    }
+
+
+Module parameters
+-----------------
+
+A module script can define parameters using the same syntax as a Nextflow workflow script::
+
+    params.foo = 'Hello'
+    params.bar = 'world!'
+
+    def sayHello() {
+        println "$params.foo $params.bar"
+    }
+
+
+When including a module, the module will first use parameters from the including context. For example::
+
+    params.foo = 'Hola'
+    params.bar = 'Mundo'
+
+    include { sayHello } from './some/module'
+
+    workflow {
+        sayHello()
+    }
+
+The above snippet prints::
+
+    Hola Mundo
+
+.. note::
+    The module inherits the parameters defined *before* the ``include`` statement, therefore any parameters
+    set afterwards will not be used by the module.
+
+.. tip::
+    It is best to define all pipeline parameters *before* any ``include`` statements.
+
+The ``addParams`` option can be used to pass parameters to the module without affecting the including
+scope.
```

Comment:
I find this wording a little awkward, but can't think of a better way to say it 🤔

Reply:
I will give it a try :smile: